.githooks/pre-commit

```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
+
+# Move to the git repository root
+REPO_ROOT=$(git rev-parse --show-toplevel)
+cd "$REPO_ROOT"
+
+# Stash non-staged changes to avoid linting them
+git stash push -q --keep-index
```

Comment:
You could use `trap 'git stash pop -q' EXIT INT TERM` to automatically pop when the script exits.